### PR TITLE
FISH-5890 Upgrade MicroProfile Rest Client to 3.0

### PR DESCRIPTION
**Pull request by jGauravGupta**
Signed-off-by: Gaurav Gupta <gaurav.gupta@payara.fish>

### CODE REVIEW

**Comment by jansupol** [comment]
The actual branch, 3.1 already contains the upgrade to MP Client 3.0.

**Comment by jansupol** [comment]
Jersey 3.1.0-M1 in staging contains the implementation https://jakarta.oss.sonatype.org/content/groups/staging/org/glassfish/jersey/ext/microprofile/jersey-mp-rest-client/3.1.0-M1/

File: ext/microprofile/mp-rest-client/pom.xml [hunk 1]
```diff
@@ -110,7 +109,7 @@
                 <configuration>
                     <instructions>
                         <Export-Package>
-                            org.glassfish.jersey.restclient.*;version=${project.version}
+                            org.glassfish.jersey.microprofile.restclient.*;version=${project.version}
```
Comment (jansupol) [inline ext/microprofile/mp-rest-client/pom.xml:112]: We have this wrong in 3.1, I assume

Comment (jansupol) [reply ext/microprofile/mp-rest-client/pom.xml:112]: We are planning 3.1.0-M2 this week, we should fix it there.

File: pom.xml [hunk 2]
```diff
@@ -2081,7 +2081,8 @@
 
         <bouncycastle.version>1.68</bouncycastle.version>
         <commons-lang3.version>3.3.2</commons-lang3.version>
-        <microprofile.config.version>2.0</microprofile.config.version>
+        <microprofile.config.version>3.0.1</microprofile.config.version>
```
Comment (jansupol) [inline pom.xml:2084]: We probably should not update the config version to 3.x, there is no implementation for it out there, yet.

File: ext/microprofile/mp-rest-client/pom.xml (hunk 1 above)
Comment (jansupol) [reply ext/microprofile/mp-rest-client/pom.xml:112]: #4947

**Comment by jGauravGupta** [comment]
Thanks @jansupol , As 3.1 branch already have this upgrade, closing this PR.

**Pull request closed without merging**